### ISSUE TITLE
rflash reports Error: timeout and then success.

### ISSUE THREAD

**Issue by ralphbellofatto** [xcat2/xcat-core#2682]
we have two nodes what are currently not responding to the BMC.

When i attempt to use rflash on them, we see:
```
c460c010: Error: timeout
c460c003: Error: timeout
c460c003: success to update firmware
c460c010: success to update firmware
```

This is picitulary problematic when we are attempting to use some automation on large numbers of nodes to determine that things went well...

for example:
```

c460c017: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c017.log"
c460c006: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c006.log"
c460c010: Error: timeout
c460c003: Error: timeout
c460c003: success to update firmware
c460c010: success to update firmware
c460c001: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c001.log"
c460c004: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c004.log"
c460c008: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c008.log"
c460c002: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c002.log"
c460c012: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c012.log"
c460c015: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c015.log"
c460c013: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c013.log"
c460c005: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c005.log"
c460c007: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c007.log"
c460c018: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c018.log"
c460c014: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c014.log"
c460c009: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c009.log"
c460c011: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c011.log"
c460c016: rflashing ... See the detail progress : "tail -f /var/log/xcat/rflash/c460c016.log"
c460c001: success to update firmware
```

**Comment by whowutwut** [comment]
@chenglch  Is this just a case where we are returning the wrong message and not detecting the correct return code from the `Error: timeout` and printing "success to update firmware" or are we actually getting a success return code and then printing that message.

**Comment by chenglch** [comment]
@whowutwut   This issue hit the timeout event in the `donode` subroutine which is a part of the `perl version code` to establish the ipmi session. If the command is `rflash`, we start processes to call these subroutine, then get the necessary information that the ipmitool-xcat command needed.  But the old `perl version ipmi code` do not return error(subroutine level, as it should maintain a select loop) when returned from our `rflash` child processes.  So my fix will not to check the status of child processes and send the error or success message directly from the child processes.

**Comment by zet809** [comment]
Hi, @ralphbellofatto , we had fixed this defect and #2693 with pull request #2725 , it had been included in our 2.13.3 RC1 build (available at https://xcat.org/files/xcat/xcat-core/2.13.x_Linux/core-snap/20170330.2215-core-rpms-snap.tar.bz2), will you pls verify whether the fix works for you? Thx a lot!

**Comment by whowutwut** [comment]
@gurevichmark  has been running the testing on r92, so he can validate and close this issue on Monday.
